ci/release/verify_release.py

```diff
@@ -0,0 +1,90 @@
+#!/usr/bin/env python
+
+from __future__ import annotations
+
+import argparse
+import ast
+import concurrent.futures
+import itertools
+import sys
+from functools import partial
+from pathlib import Path
+from typing import NamedTuple
+
+from packaging import version as v
+
+
+class RemovedIn(NamedTuple):
+    version: v.Version
+    lineno: int
+
+
+class RemovedInVisitor(ast.NodeVisitor):
+    def __init__(self) -> None:
+        super().__init__()
+        self.removed_ins: list[RemovedIn] = []
+
+    def visit_keyword(self, node: ast.keyword) -> None:
+        if node.arg == "removed_in":
+            value = node.value
+            if isinstance(value, ast.Constant):
+                self.removed_ins.append(
+                    RemovedIn(
+                        version=normalize_version(value.value), lineno=node.lineno
+                    )
+                )
+
+
+def normalize_version(vstr: str) -> v.Version:
+    ver = v.parse(vstr)
+    return v.parse(f"{ver.major}.{ver.minor}.{ver.micro}")
+
+
+def find_invalid_removed_ins(
+    next_ver: v.Version, path: Path
+) -> list[tuple[Path, RemovedIn]]:
+    code = path.read_text()
+    node = ast.parse(code)
+    visitor = RemovedInVisitor()
+    visitor.visit(node)
+    return [
+        (path, removed_in)
+        for removed_in in visitor.removed_ins
+        if removed_in.version <= next_ver
+    ]
+
+
+def main(next_ver: v.Version, root: str) -> int:
+    pyfiles = root.rglob("*.py")
```

Comment:
```suggestion
def main(next_ver: v.Version, root: Path) -> int:
    pyfiles = root.rglob("*.py")
```
if you're calling `rglob` this can only be a `pathlib.Path`, I think